dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageForm.java

```diff
@@ -0,0 +1,102 @@
+package com.dotcms.rest.api.v1.page;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.dotcms.contenttype.transform.JsonTransformer;
+import com.dotcms.repackage.com.fasterxml.jackson.annotation.JsonProperty;
+import com.dotcms.repackage.com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.dotcms.repackage.javax.validation.constraints.NotNull;
+import com.dotcms.rest.exception.BadRequestException;
+import com.dotmarketing.portlets.templates.design.bean.TemplateLayout;
+import com.dotmarketing.util.json.JSONException;
+import com.dotmarketing.util.json.JSONObject;
+
+/**
+ * {@link PageResource}'s form
+ */
+@JsonDeserialize(builder = PageForm.Builder.class)
+public class PageForm {
+
+    private String themeId;
+    private String title;
+    private String hostId;
+    private TemplateLayout layout;
+
+    public PageForm(String themeId, String title, String hostId, TemplateLayout layout) {
+        this.themeId = themeId;
+        this.title = title;
+        this.hostId = hostId;
+        this.layout = layout;
+    }
+
+    public String getThemeId() {
+        return themeId;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getHostId() {
+        return hostId;
+    }
+
+    public TemplateLayout getLayout() {
+        return layout;
+    }
+
+    public static final class Builder {
+        @JsonProperty
+        @NotNull
+        private String themeId;
+
+        @JsonProperty
+        private String title;
+
+        @JsonProperty
+        @NotNull
+        private String hostId;
+
+        @JsonProperty
+        @NotNull
+        private Map<String, Object> layout;
+
+        public Builder() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)